packages/corejs/src/monitoring/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { createContext, useContext, useEffect, useMemo } from 'react';
+import { Primitive } from '@sentry/types';
+import { CaptureConsole } from '@sentry/integrations';
+import { Integrations } from '@sentry/tracing';
+import { Platform } from 'react-native';
+import * as Sentry from 'sentry-expo';
+import * as SentryR from '@sentry/react';
+import Constants from 'expo-constants';
+import PropTypes from 'prop-types';
+
+import { MonitoringContextType } from './types';
+
+const tracingOrigins = ['localhost', 'cna.dev.monk.ai', 'cna.staging.monk.ai', 'cna.preview.monk.ai', 'cna.monk.ai'];
+const MonitoringConstants = {
+  type: {
+    UPLOAD: 'upload', // logs linked to the upload
+    CAMERA: 'camera', // logs linked to the camera
+    FUNC: 'func', // logs linked to a function
+    APP: 'app', // logs linked to the application
+    HTTP: 'http', // logs linked to the api
+  },
+  operation: {
+    HTTP: 'http',
+    USER_TIME: 'user-time-per-action',
+    USER_CAMERA_TIME: 'user-camera-time',
+    USER_UPLOAD_CENTER_TIME: 'user-upload-center-time',
+    USER_ACTION: 'user-action',
+    RESPONSE_TIME: 'response-time',
+    FUNC: 'func',
+  },
+};
+
+const Tracing = Integrations.BrowserTracing;
+const setTag = (key: string, value: Primitive): void => SentryR.setTag(key, value);
+const setUser = (id: string) => SentryR.setUser({ id });
+const MonitoringContext = createContext<MonitoringContextType | null>(null);
+
+function MonitoringProvider({ children, config }) {
+  useEffect(() => {
+    Sentry.init({
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      dsn: (config.dns || Constants.manifest.extra.SENTRY_DSN) as string,
+      environment: Constants.manifest.extra.ENV as string,
+      debug: Constants.manifest.extra.ENV !== 'production',
+      enableAutoSessionTracking: true,
+      enableInExpoDevelopment: true,
+      sessionTrackingIntervalMillis: 10000,
+      tracesSampleRate: Constants.manifest.extra.ENV !== 'production' ? 1.0 : 0.2,
```

Comment:
All of this configuration object should not be a magic value and should be defined as a const somewhere